pkg/database/postgresql.go

```diff
@@ -66,50 +66,50 @@ type PostgreSQLFilter struct {
 }
 
 func (PostgreSQLFilter) CreationTSAndIDFilter(cond sqlbuilder.Cond, continueDate, continueId string) string {
-	return cond.Var(sqlbuilder.Build(
-		"(data->'metadata'->>'creationTimestamp', id) < ($?, $?)",
-		continueDate, continueId,
-	))
+	return fmt.Sprintf(
+		"(data->'metadata'->>'creationTimestamp', id) < (%s, %s)",
+		cond.Var(continueDate), cond.Var(continueId),
+	)
 }
 
 func (PostgreSQLFilter) OwnerFilter(cond sqlbuilder.Cond, owners []string) string {
-	return cond.Var(sqlbuilder.Build(
-		"jsonb_path_query_array(data->'metadata'->'ownerReferences', '$[*].uid') ?| $?",
-		pq.Array(owners),
-	))
+	return fmt.Sprintf(
+		"jsonb_path_query_array(data->'metadata'->'ownerReferences', '$[*].uid') ?| %s",
+		cond.Var(pq.Array(owners)),
+	)
 }
 
 func (PostgreSQLFilter) ExistsLabelFilter(cond sqlbuilder.Cond, labels []string) string {
-	return cond.Var(sqlbuilder.Build(
-		"data->'metadata'->'labels' ?& $?",
-		pq.Array(labels),
-	))
+	return fmt.Sprintf(
+		"data->'metadata'->'labels' ?& %s",
+		cond.Var(pq.Array(labels)),
+	)
 }
 
 func (PostgreSQLFilter) NotExistsLabelFilter(cond sqlbuilder.Cond, labels []string) string {
-	return cond.Var(sqlbuilder.Build(
-		"NOT data->'metadata'->'labels' ?| $?",
-		pq.Array(labels),
-	))
+	return fmt.Sprintf(
+		"NOT data->'metadata'->'labels' ?| %s",
+		cond.Var(pq.Array(labels)),
+	)
 }
 
-func (PostgreSQLFilter) EqualsLabelFilter(cond sqlbuilder.Cond, labels map[string]string) string {
+func (PostgreSQLFilter) EqualsLabelFilter(_ sqlbuilder.Cond, labels map[string]string) string {
 	jsonLabels, _ := json.Marshal(labels)
-	return cond.Var(sqlbuilder.Build(
-		"data->'metadata'->'labels' @> $?",
+	return fmt.Sprintf(
+		"data->'metadata'->'labels' @> %s",
 		string(jsonLabels),
-	))
+	)
```

Comment:
@ggallen this is the only change that is "open" to sql injection, I'm changing it so we use it for the jsonlabels var

Reply:
fixed!